src/Mvc/Mvc.Razor.RuntimeCompilation/src/AssemblyPartResolvers.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Mvc.ApplicationParts;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation
+{
+    /// <summary>
+    /// This class is a workaround to enable compilation of plugin references which DependencyContext is unaware of
+    /// </summary>
+    internal class DependencyContextResolver : IAssemblyPartResolver
+    {
+        public IEnumerable<string> GetReferencePaths(AssemblyPart assemblyPart)
+        {
+            try
+            {
+                return assemblyPart.GetReferencePaths();
+            }
+            catch (InvalidOperationException)
```

Comment:
We specifically want this to throw if this fails. Suppressing this error is not something we want.

Reply:
I suppose there has to be another way around this somehow, but the given code can't be called for external libraries or otherwise any change to an unrelated .cshtml breaks runtime compilation entirely, which renders the runtime compilation project useless for any project that references external libraries. As it is for 3.1 i will change code further while avoiding excessive changes

Reply:
I've added another method on the interface to provide a way of resolving the resolution of an assembly part in a way where it is possible to consider an assembly part resolved without going through the DependencyContextResolver if it has been processed already. Also the exception is back in place with additional info on how to resolve the exception it raises.